src/transformStateWithClones.js

```diff
@@ -8,6 +8,28 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
-}
+  const result = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
```

Comment:
Move all string values to constants to avoid errors connected with typos  in the future